Space Pirates

FlattestGuitar opened the pull request:
:cl: FlattestGuitar
rscdel: absolutely kills Vox Raiders as a gamemode
rscadd: adds Raiders as a gamemode with more rum and less skrek
rscadd: adds the AP10, a 10mm machine pistol, only available to Raiders
rscadd: adds a swanky space pirate Plasmaman suit, again, only Raiders
/:cl:

Ooookay, so:
Considering Vox Raiders is widely considered to be an incredibly flawed gamemode and has been taken out of the standart rotation without much protest, I've decided to simply work on top of it, pretty much nuking the thing in favour of THIS. If that's an issue, I'll gladly redo all of the work I've done in another spot, so the old gamemode is still accessible, as it's kinda my fault for not asking in the first place.

![](https://i.gyazo.com/a2a11abfb55a0a57ab8f73cc027cc76e.png)

This PR also adds a new weapon!
A 10mm machine pistol - the AP10
two round burst, 24+1 ammo, 30 damage per shot, quite powerful
![](https://i.gyazo.com/9d1562ab6f28a207b12c19819111420e.png)

Also adds this neat plasmaman suit
![](https://i.gyazo.com/2de3861ef47d735afb0d2bc8b2858aa2.png)

Here's a list of features that are already in or are intended to be in eventually, in no particular order:
- All raiders KEEP their selected species, getting a random name and appearance. This spices things up and makes for more interesting RP.
- All raiders get a normal survival box with two additional 20u synthflesh patches inside.
- All raiders get a random space suit and a random firearm with a reasonable supply of ammo. If their firearm is crap, they also get a melee weapon to improve their chances.
- The raiders get a large amount of random stuff that can be valuable to the crew. Materials, tech disks, pizza, fancy hats, things that will make it possible to actually trade with them.
- Raiders get objectives that can either be objectively antagonistic (kidnap, assasinate) or can be reached through trade and cooperation with the crew. Attacking them preemptively because of the metagamey "They have to hurt us eventually" won't work. That said, if the raiders need to get an emitter, they might as well slice and shoot their way to it, if they please.
- The raider crews can be anywhere from 3 to 6 people, so as not to have negotiations along the lines of "There's always five of you, why did only four come?"

TO DO:
- [x] Actually make this keeps your species, not make you a human 100% of the time. How did this get lost?
- [x] Rip out all mentions of Vox Raiders instead of just Raiders
- [x] Make kidnap and assasinate not target SNPCs. How did the other PR do that, again?
- [x] Add in one_click_antag functionality, which I merrily ripped out because it was ugly
- [x] Add species handling for new raiders. Vox need internals, Plasmamen need suits
- [x] Properly remap the Raider Den and Shuttle
- [x] Add in spawners for Raider gear and trade stuff
- [x] Add new funky sprites for a few things, especially uniforms and A SPACE PIRATE PLASMAMAN SUIT
- [x] Find slaves to do those for me
- [x] Figure out what cool stuff the raiders should spawn with that would make them both look cool and be functional
- [x] Cry a bit, because this is long

BECAUSE I CHANGED ROLE DEFINES, THIS WILL NEED THE DB MODIFIED. I don't know how to do that, so someone else will have to figure out the query.

*also, this is a reopen of #5625 because ~~I can't use Git~~ Git is absolutely bugged in every way and this is not my fault. rip.*

taukausanake commented:
I would call Feature Freeze on this because it is a PR that was made after such. :P But yeah, reopen.

Git gud :^)

Spacemanspark commented:
:+1: merge immediately.

Allfd commented:
What would need to change on the SQL end?

Crazylemon64 commented:
@taukausanake It's a re-open of one that existed beforehand, it's in the clear

taukausanake commented:
@Crazylemon64 I know that, can't I be a twat about losing skree raiders? I'd really like this to be specie raiders/pirates including as it is right now with random species per member. A full group of Plasmamen or Diona pirates would be cool, let alone Vox

Spacemanspark commented:
Diversity is fun and makes for interesting situations.

Crazylemon64 commented on code/game/gamemodes/objective.dm:
```diff
@@ -517,20 +517,21 @@ var/list/potential_theft_objectives = subtypesof(/datum/theft_objective) \
 /datum/objective/heist/kidnap
 /datum/objective/heist/kidnap/choose_target()
-	var/list/roles = list("Chief Engineer","Research Director","Roboticist","Chemist","Station Engineer")
+	var/list/roles = list("Head of Security", "Head of Personnel", "Captain", "Nanotrasen Representative")
```
Any reason for changing this from technicians to command staff? To increase the challenge of the objective, or...?

FlattestGuitar replied:
Yeah. I think the raiders would have it too easy if they were to have to kidnap a less important person. Considering our player numbers, I think it's safe to assume we'll have one of these guys on the station.

Crazylemon64 commented on code/game/gamemodes/objective.dm:
```diff
@@ -636,99 +671,31 @@ var/list/potential_theft_objectives = subtypesof(/datum/theft_objective) \
+			return 1
+		if(!target.current.ckey)
+			return 1
+		return 0
```
`assasinate` should be spelled `assassinate`

FlattestGuitar replied:
addressed

FlattestGuitar commented:
I just want to point something out: this is VERY experimental and I really don't expect the gamemode to work well design-wise as it stands. There's still a lot to tweak and work on in the future and this PR is just a foundation to what will come.

I have a few ideas in store, but getting more input is always good.

Fox-McCloud commented:
Oh boy, *icon conflicts*.

Those are ***so much fun***.

monster860 commented:
Use dmitool.

Iamgoofball commented:
@Fox-McCloud close for feature freeze when?

Fox-McCloud commented:
@Iamgoofball 

This PR was submitted before the freeze; he merely re-submitted it.

SamHPurp commented:
Looks good! (Gimme dat Plasmasuit)

Crazylemon64 commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -103,7 +103,7 @@
 				/obj/item/weapon/paper/crumpled = 10,
 				/obj/item/weapon/pen = 10,
-				 /obj/item/weapon/minihoe = 10,
+				/obj/item/weapon/minihoe = 10,
```
I think this is a "cultivator" now or something?

Crazylemon64 commented on code/modules/admin/verbs/one_click_antag.dm:
```diff
@@ -529,6 +426,58 @@ client/proc/one_click_antag()
+						return
+					else
+						return
+	sleep(300)
```
Any reason `poll_candidates` or similar is not in use here?

Crazylemon64 commented on code/modules/admin/verbs/one_click_antag.dm:
```diff
@@ -529,6 +426,58 @@ client/proc/one_click_antag()
+	var/input = "Loot and burn the station or trade with it."
+
+	for(var/mob/G in respawnable_list)
+		if(!jobban_isbanned(G, "Raider"))
```
You should use `SPECIAL_ROLE_RAIDER` here to ward from typos in the future

FlattestGuitar commented:
I believe I've addressed all issues.

I'll be making a new PR that changes up admin outfits to toss out the underused ones and make the ones we have more interesting. I'll tweak the admin raider outfit in there, as well as making the Vox raider outfits (should they ever be used by anyone) working.

Merge when.

Spacemanspark commented:
Merge at once
immediately

scrubmcnoob commented:
I'm waiting for that delicious Tec-9

Crazylemon64 commented on code/modules/admin/verbs/one_click_antag.dm:
```diff
@@ -528,6 +425,48 @@ client/proc/one_click_antag()
+	for(var/mob/G in respawnable_list)
+		if(!jobban_isbanned(G, SPECIAL_ROLE_RAIDER))
+			spawn(0)
+				candidates = pollCandidates("Do you wish to be considered for a Space Pirate party about to be sent in?", SPECIAL_ROLE_RAIDER, 1)
```
This is going to ask for candidates many times for the same person. Also, since the assignment to `candidates` is in a `spawn`, you can't use the returned value. You can just get rid of all the conditionals, spawn, and looping, and you'll be just fine.

FlattestGuitar replied:
What in the world did I do here...

Aracino commented:
This cant diiiie

monster860 commented:
Ftl13 has had space pirates since the beginning

tigercat2000 commented on code/datums/outfits/outfit_admin.dm:
```diff
@@ -560,54 +560,17 @@
 
-
-
-/datum/outfit/admin/vox
```
please do not remove this, it would be preferable that you simply remove the
```
else
 equip_vox_raider()
 regenerate_icons()
```
call from /equip. This loadout is specifically for armalises.

tigercat2000 commented on code/game/gamemodes/heist/heist.dm:
```diff
@@ -35,13 +34,12 @@ var/global/list/obj/cortical_stacks = list() //Stacks for 'leave nobody behind'
-		raider_num = recommended_enemies
+
+
+	raider_num = min(3 + round((num_players()-30)/20), 6)//3 at 30 players, plus 1 per every 20. Not more than 6
```
spacing please, all of the operators should have a space on either side of em

tigercat2000 commented on code/game/gamemodes/heist/heist.dm:
```diff
@@ -57,187 +55,186 @@ var/global/list/obj/cortical_stacks = list() //Stacks for 'leave nobody behind'
+	var/index = 1
+	for(var/datum/mind/raider in raiders)
+		if(index > raider_spawn.len)
+			index = 1
```
rather than this complex loop you could just do
```
var/index = 1
for(var/datum/mind/raider in raiders)
    raider.current.loc = raider_spawn[index % (raider_spawn.len + 1)]
    make_raider_gear(raider_gear[index % (raider_spawn.len + 1)])
    index++
```

tigercat2000 commented on code/game/gamemodes/heist/heist.dm:
```diff
@@ -57,187 +55,186 @@ var/global/list/obj/cortical_stacks = list() //Stacks for 'leave nobody behind'
+	M.rejuvenate()//removes people being stupid and taking limbless, mute and blind characters, also regenrates icons.
+
+/datum/game_mode/heist/proc/make_raider_gear(var/location)
+	if(!istype(location, /turf))
```
`if(!isturf(location))`

tigercat2000 commented on code/modules/admin/verbs/one_click_antag.dm:
```diff
@@ -528,6 +425,45 @@ client/proc/one_click_antag()
+			candidates.Remove(theghost)
+			var/mob/living/carbon/human/M = new(get_turf(L))
+			M.key = theghost.key
+			var/datum/preferences/P = new
```
why is this different than the normal make raider proc

tigercat2000 commented on data/mode.txt:
```diff
@@ -1 +1 @@
-extended
+secret
```
don't change this file

tigercat2000 commented on code/modules/admin/verbs/vox_raiders.dm:
```diff
@@ -1,65 +0,0 @@
-	R.set_frequency(SYND_FREQ) //Same frequency as the syndicate team in Nuke mode.
-	equip_to_slot_or_del(R, slot_l_ear)
-
-	equip_to_slot_or_del(new /obj/item/clothing/under/vox/vox_robes(src), slot_w_uniform)
```
I would really rather you don't just kill off this proc entirely and that you add all of these loadouts as actual admin loadouts, these aren't useless

SheepFighter commented:
Arrrr! Be this comin' soon? We Arrrrsstafarrrrrrrians be wantin' new ways to be followin' the teachins' of our Lord and Captain, Blackbeard the Ascended Master!

Er...
What was I saying now?
What's the progress on this? Is there anything resembling an ETA?

FlattestGuitar commented:
I've thought about this for a long time, and I really don't think Space Pirates have what it takes to be an interesting gamemode. The cross between Nukies and Vox Raiders makes them bad at both things, not in any way unique or fun.

I've ran numerous events that aimed at emulating what this PR is supposed to implement and ALL of them ended with a messy shootout which had little sense.

There's possibly a way to make this into an interesting gamemode, but I just can't see it, and considering I have other stuff to work on currently, I don't see much hope for Space Pirates.

This is not because the Maints didn't want to merge this PR, this is purely because I feel the gamemode just won't work on Paradise.

I'M SORRY, OKAY!?

Rapidvalj commented:
😭

taukausanake commented:
Does this mean we still have Voxes? I don't care how shitty they are, I like seeing those birbs fucking about on the station but they really shouldn't be a gamemode, just event

Fox-McCloud commented:
Aww, that's too bad, but I can easily see happening what you described. Welp....Rest in pieces, I guess. Glad you got to see the merits (or not), before you had to put more work into it; not all of us get the advantage of knowing, for 100% certainty, if something is completely a good idea or not.
